src/anim/evaluator/anim-target-value.js

```diff
@@ -58,17 +85,37 @@ class AnimTargetValue {
     updateValue(index, value) {
         // always reset the value of the target when the counter is 0
         if (this.counter === 0) {
-            this.value[0] = 0;
-            this.value[1] = 0;
-            this.value[2] = 0;
-            this.value[3] = 1;
+            AnimEvaluator._set(this.value, [0, 0, 0, 1], this.valueType);
+            if (!this._normalizeWeights) {
+                AnimEvaluator._blend(this.value, this.baseValue, 1, this.valueType);
+            }
         }
         if (!this.mask[index]) return;
-        if (this.counter === 0) {
-            AnimEvaluator._set(this.value, value, this.valueType);
+        if (this._layerBlendType(index) === ANIM_LAYER_ADDITIVE && !this._normalizeWeights) {
+            if (this.valueType === AnimTargetValue.TYPE_QUAT) {
+                // current value
+                const v = AnimTargetValue.q1.set(...this.value);
+                // additive value
+                const aV = AnimTargetValue.q2.set(...this.baseValue).invert().mul(AnimTargetValue.q3.set(...value));
+                // scale additive value by it's weight
+                aV.slerp(Quat.IDENTITY, aV, this.getWeight(index));
+                // add the additive value onto the current value then set it to the targets value
+                v.mul(aV);
+                AnimEvaluator._set(this.value, [v.x, v.y, v.z, v.w], this.valueType);
```

Comment:
any good way to implement this without array allocation? Maybe have preallocate array and just set values to it.

Reply:
I've moved this to a static array.